Program.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Pomelo.EntityFrameworkCore.MySql.Infrastructure;
+using System;
+using System.Text;
+using Newtonsoft.Json;
+using Microsoft.AspNetCore.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddDbContext<ApplicationDbContext>(options =>
+{
+    var connectionString = "Server=localhost;Database=myappdb;User=myappuser;Password=mypassword;";
+    Console.WriteLine($"Connection String: {connectionString}");
+    options.UseMySql(connectionString, ServerVersion.AutoDetect(connectionString));
+});
+
+// Move this line before builder.Build();
+builder.Services.AddHealthChecks();
+
+var app = builder.Build();
+
+app.UseHttpsRedirection();
+app.UseRouting();
+
+app.UseEndpoints(endpoints =>
+{
+    endpoints.MapHealthChecks("/health", new HealthCheckOptions

```

Comment:
Health checks in C# is WAY simpler. You don't need a third party library or anything: `app.UseHealthChecks("/health");`

P.S. You might want to read a bit about C# Asp.Net Middleware Pipeline.

Because the order of .use / .map makes a slight difference